src/map_object.c

```diff
@@ -107,493 +119,472 @@ LocalMapObject* sub_0205E1D0(MapObjectManager* manager, ObjectEvent* object_even
+    MapObject_SetType(localObject, savedObject->type);
+    MapObject_SetFlagID(localObject, savedObject->flagId);
+    MapObject_SetScript(localObject, savedObject->script);
+    MapObject_SetInitialFacingDirection(localObject, savedObject->initialFacing);
+    MapObject_SetFacingDirectionDirect(localObject, savedObject->currentFacing);
+    MapObject_SetNextFacingDirection(localObject, savedObject->nextFacing);
+    MapObject_SetParam(localObject, savedObject->param[0], 0);
+    MapObject_SetParam(localObject, savedObject->param[1], 1);
+    MapObject_SetParam(localObject, savedObject->param[2], 2);
+    MapObject_SetXRange(localObject, savedObject->xRange);
+    MapObject_SetYRange(localObject, savedObject->yRange);
+    MapObject_SetInitialX(localObject, savedObject->initialX);
+    MapObject_SetInitialHeight(localObject, savedObject->initialHeight);
+    MapObject_SetInitialY(localObject, savedObject->initialY);
+    MapObject_SetCurrentX(localObject, savedObject->currentX);
+    MapObject_SetCurrentHeight(localObject, savedObject->currentHeight);
+    MapObject_SetCurrentY(localObject, savedObject->currentY);
+
+    VecFx32 coords = {};
+    coords.y = savedObject->vecY;
+    MapObject_SetPositionVec(localObject, &coords);
+
+    memcpy(sub_0205F394(localObject), savedObject->unk30, 16);
+    memcpy(sub_0205F3BC(localObject), savedObject->unk40, 16);
+}
+
+static void sub_0205E8EC(MapObjectManager *manager, LocalMapObject *object) {
     sub_0205E934(object);
-    sub_0205E954(object);
-    sub_0205F354(object, manager);
+    MapObject_ConvertXYToPositionVec(object);
+    MapObject_SetManager(object, manager);
     sub_0205ECE0(object);
     MapObject_ClearHeldMovement(object);
     sub_0205EFB4(object);
     sub_0205EAF0(manager, object);
     sub_0205F450(object);
-    sub_0205F16C(sub_0205F160(manager));
-}
-
-static void sub_0205E934(LocalMapObject* object) {
-    MapObject_SetFlagsBits(object, MAPOBJECTFLAG_UNK2 | MAPOBJECTFLAG_ACTIVE);
-    MapObject_ClearFlagsBits(object, MAPOBJECTFLAG_IGNORE_HEIGHTS |
-                                     MAPOBJECTFLAG_UNK22 |
-                                     MAPOBJECTFLAG_UNK21 |
-                                     MAPOBJECTFLAG_UNK19 |
-                                     MAPOBJECTFLAG_UNK18 |
-                                     MAPOBJECTFLAG_UNK17 |
-                                     MAPOBJECTFLAG_UNK16 |
-                                     MAPOBJECTFLAG_UNK14 |
-                                     MAPOBJECTFLAG_VISIBLE |
-                                     MAPOBJECTFLAG_MOVEMENT_PAUSED |
-                                     MAPOBJECTFLAG_UNK3);
+    sub_0205F16C(MapObjectManager_GetMapObjectManager(manager));
+}
+
+static void sub_0205E934(LocalMapObject *object) {
+    MapObject_SetFlagsBits(object, (MapObjectFlagBits)(MAPOBJECTFLAG_UNK2 | MAPOBJECTFLAG_ACTIVE));
+    MapObject_ClearFlagsBits(object, (MapObjectFlagBits)(MAPOBJECTFLAG_IGNORE_HEIGHTS | MAPOBJECTFLAG_UNK22 |
```

Comment:
not sure how much I like the stacking compared to before

Reply:
it does kinda make sense to be an enum type, it's just flags enums are annoying, I wonder if there's a way to get the compiler to work properly with this, would be good

Reply:
as I said above, I'll probably change this to a define in the next PR (cleanup 2, whenever that happens)